lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -174,6 +174,21 @@ Type CompleteGenericTypeResolver::resolveDependentMemberType(
     return DependentMemberType::get(baseTy, assocType, TC.Context);
   }
 
+  // If the nested type comes from a type alias, use either the alias's
+  // concrete type, or resolve its components down to another dependent member.
+  if (auto alias = nestedPA->getTypeAliasDecl()) {
+    if (nestedPA->isConcreteType())
+      return nestedPA->getConcreteType();
+
+    auto repr = alias->getUnderlyingTypeLoc().getTypeRepr();
+    if (auto identRepr = dyn_cast<IdentTypeRepr>(repr)) {
+      for (auto component : identRepr->getComponentRange())
+        baseTy = resolveDependentMemberType(baseTy, DC, baseRange, component);
```

Comment:
Can't use lookupMemberType(), unfortunately, as it specifically asserts if you try to call it with a dependent type...
```
  // Callers must cope with dependent types directly.  
  assert(!type->isTypeParameter());
```